src/citrine/resources/material_spec.py

```diff
@@ -87,3 +88,28 @@ class MaterialSpecCollection(DataConceptsCollection[MaterialSpec]):
     def get_type(cls) -> Type[MaterialSpec]:
         """Return the resource type in the collection."""
         return MaterialSpec
+
+    def with_template(self,
```

Comment:
Thoughts on calling this `filter_by_template`?

Reply:
I am on the side of keeping this consistent by using filter_by_template.